### PR TITLE
Add code and info to mongoerror

### PR DESCRIPTION
**Pull request by harish-chandramowli**
1 - Remove interface to persist client message 
2 - Allow access to mongo error code

### CODE REVIEW

File: inttests/int_test_utils.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ func insertDummyDocs(client *mongo.Client, numOfDocs int, ctx context.Context) e
 	if _, err := coll.InsertMany(ctx, docs); err != nil {
 		return fmt.Errorf("initial insert failed. err: %v", err)
 	}
+	fmt.Println("inserted", numOfDocs, "dummy docs")
```
Comment (TomerYakir) [inline inttests/int_test_utils.go:35]: [q] stray debug message?

File: bsonutil.go [hunk 2]
```diff
@@ -138,6 +138,26 @@ func GetAsBSON(elem bson.E) (bson.D, string, error) {
...
+	case primitive.A:
+		res := make([]string, len(val))
+		for num, raw := range []interface{}(val) {
+			switch fixed := raw.(type) {
+			case string:
+				res[num] = fixed
+			default:
+				return nil, tipe, NewStackErrorf("not string %T %s", raw, raw)
+			}
+			return res, tipe, nil
```
Comment (TomerYakir) [inline bsonutil.go:153]: [q] are we returning after one iteration?

Comment (harish-chandramowli) [reply bsonutil.go]: Its a bug. i added the test for this now

**Comment by harish-chandramowli** [comment]
@TomerYakir : RFAL

**Review by TomerYakir — APPROVED** [review]
LGTM on green

**Pull request merged**